File: src/dotnet/WifiRemoteService/Services/WifiRemoteService.cs
using Grpc.Core;
using Microsoft.Net.Wifi.Remote.Service;
using System.Text;

using grpc = global::Grpc.Core;

namespace Microsoft.Net.Wifi.Remote.Service
{
    public class WifiRemoteService : WifiRemote.WifiRemoteBase
    {
        private readonly ILogger<WifiRemoteService> _logger;

        public WifiRemoteService(ILogger<WifiRemoteService> logger)
        {
            _logger = logger;
        }

        public override Task<Microsoft.Net.Wifi.Remote.Response> GetWifiApControl(Microsoft.Net.Wifi.Remote.Request request, grpc::ServerCallContext context)
        {
            return Task.FromResult(new Microsoft.Net.Wifi.Remote.Response
            {
                RequestId = request.Id,
                Payload = Google.Protobuf.ByteString.CopyFromUtf8("Pong")
            });
        }
    }
}

File: src/service/WifiRemoteService.cxx

#include <format>
#include <iostream>

// #include "include/microsoft/wifi/remote/WifiRemoteService.hxx"
#include <microsoft/wifi/remote/WifiRemoteService.hxx>

using namespace Microsoft::Net::Wifi::Remote::Service;

::grpc::Status WifiRemoteService::GetWifiApControl([[maybe_unused]] ::grpc::ServerContext* context, const ::Microsoft::Net::Wifi::Remote::Request* request, ::Microsoft::Net::Wifi::Remote::Response* response)
{
    std::cout << std::format("Received GetWifiApControl request with id {} and payload {}", request->id(), request->payload()) << '\n';

    response->set_requestid(request->id());
    response->set_payload("Pong");

    return grpc::Status::OK;
}
